sphinx/builders/linkcheck.py

```diff
@@ -460,7 +460,12 @@ def check_uri() -> Tuple[str, str, int]:
                                                  config=self.config, auth=auth_info,
                                                  **kwargs)
                         response.raise_for_status()
-                    except (HTTPError, TooManyRedirects) as err:
+                    # When there is a ConnectionError from the HEAD request, it might be due to
+                    # the webserver intentionally dropping the connection when it sees HEAD requests
+                    # but still responding properly to GET requests. This has been observed in the
+                    # wild with at https://patft.uspto.gov/.
+                    # See https://github.com/sphinx-doc/sphinx/issues/9306 for more details.
```

Comment:
The details are available through git blame, just a short comment is enough.
```suggestion
                    # Servers drop the connection on HEAD requests, causing
                    # ConnectionError.
```